Feature/frontend/job posting

niichapats opened the pull request:
## Title
Add Job Posting Page with real-time preview and company integration

## Description
- Created a new **Job Posting Page** UI for employer users.  
  - Includes three main sections:  
    - **CompanyProfileCard** — displays key company details fetched from backend API.  
    - **JobPostingForm** — interactive form for creating new job posts (title, description, qualifications, etc.).  
    - **JobPostPreview** — real-time preview panel that updates dynamically as the employer fills out the form.  
- Integrated company data via API to auto-populate company information on the page.  
- Connected the form submission to backend API for job posting creation (`POST /api/job`).  
- Added validation, form feedback, and error handling for API submission states.  

## Functional Acceptance Criteria
- [x] Company information is fetched correctly and displayed in the CompanyProfileCard.  
- [x] JobPostingForm fields update the preview in real time.  
- [x] Submitting the form calls the correct API endpoint and posts a new job successfully.  
- [x] Error and success toasts are displayed accordingly.  
- [x] Layout renders correctly.  

## Checklist
- [x] Self-review completed  
- [x] Connected form submission to backend API  
- [x] Code follows Coding Guidelines  
- [x] Verified UI state for loading, success, and error  

## Type of Change
- [x] New feature  
- [x] UI/UX enhancement  
- [x] Backend integration  

## Additional Information
- **Breaking change:** No  
- Notes:  -

SunthornK commented:
Everything’s working great overall
Just two small notes:
	•	Phone field: maybe there should be a limit it.
	•	Responsive layout: when resizing the window, a few buttons (like the calendar picker) disappear — might need a tweak for smaller breakpoints.
<img width="427" height="325" alt="image" src="https://github.com/user-attachments/assets/30c2dcf4-91f1-4cc6-a5c4-daa85d7f6820" />

niichapats commented:
> Everything’s working great overall
> Just two small notes:
> • Phone field: maybe there should be a limit it.
> • Responsive layout: when resizing the window, a few buttons (like the calendar picker) disappear — might need a tweak for smaller breakpoints.
> <img alt="image" width="427" height="325" src="https://private-user-images.githubusercontent.com/147012928/504556405-30c2dcf4-91f1-4cc6-a5c4-daa85d7f6820.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjEyNzI1MzYsIm5iZiI6MTc2MTI3MjIzNiwicGF0aCI6Ii8xNDcwMTI5MjgvNTA0NTU2NDA1LTMwYzJkY2Y0LTkxZjEtNGNjNi1hNWM0LWRhYTg1ZDdmNjgyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyNFQwMjE3MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kOThiMDhkNThmYTEyZTBiNTEyNzY0YzAxNzkxMjExMzMzOGU1NWJhNmQ1MDY4YWI0MDM3MjQ5M2ExNjVhNTE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6fpQyxaGOcQ6r3rCSr9kvT-b9ROflearWZlDPDfR22o">


Updated according to your comments. Fixed phone number length limit(15 digits) and improved layout responsiveness as requested. Also updated styles to use Lovable CSS colors and restructured code for consistency.
